internal/definition/autoarchiveexemptmetric/schema.go

```diff
@@ -0,0 +1,132 @@
+// Copyright Splunk, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package autoarchiveexemptmetric
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	automated_archival "github.com/signalfx/signalfx-go/automated-archival"
+	"go.uber.org/multierr"
+)
+
+var (
+	exemptMetricSchema = map[string]*schema.Schema{
+		"creator": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "ID of the creator of the automated archival setting",
+		},
+		"last_updated_by": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "ID of user who last updated the automated archival setting",
+		},
+		"created": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Timestamp of when the automated archival setting was created",
+		},
+		"last_updated": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Timestamp of when the automated archival setting was last updated",
+		},
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Name of the metric to be exempted from automated archival",
+		},
+	}
+)
+
+func newSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"exempt_metrics": {
+			Type:        schema.TypeList,
+			Required:    true,
+			Description: "List of metrics to be exempted from automated archival",
+			Elem: &schema.Resource{
+				Schema: exemptMetricSchema,
+			},
+			ForceNew: true,
+		},
+	}
+}
+
+func decodeTerraform(data *schema.ResourceData) (*[]automated_archival.ExemptMetric, error) {
+	exempt_metrics := &[]automated_archival.ExemptMetric{}
+	if v, ok := data.GetOk("exempt_metrics"); ok {
```

Comment:
Can I suggest inverting the condition here? The amount of nesting can be reduced significantly.